### PR TITLE
docs(vrl): Add `unflatten` vrl function documentation

### PR DESCRIPTION
**Pull request by jorgehermo9**
Relates to https://github.com/vectordotdev/vrl/pull/993

### CODE REVIEW

File: website/cue/reference/remap/functions/unflatten.cue [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	arguments: [
+		{
+			name:        "value"
+			description: "The array or object to unflatten."
+			required:    true
+			type: ["object"]
+		},
+		{
+			name:        "separator"
+			description: "The separator to be used to split flattened keys."
```
[the remark this answers is not in the document]
Comment (jorgehermo9) [reply website/cue/reference/remap/functions/unflatten.cue]: Copied that from https://github.com/vectordotdev/vector/blob/master/website/cue/reference/remap/functions/flatten.cue#L18

Should I change the definition on that file to match this one?

[the remark this answers is not in the document]
Comment (jszwedko) [reply website/cue/reference/remap/functions/unflatten.cue]: 👍 for having the descriptions be consistent. I'm ok with leaving this one as-is and following up with a PR to update both or making `flatten.cue` consistent in this PR.

**Review by jszwedko — COMMENTED** [review]
Awesome, thanks @jorgehermo9 ! I'll merge this one, as well, once VRL is upgraded in Vector.

File: website/cue/reference/remap/functions/unflatten.cue (hunk 1 above)
[the remark this answers is not in the document]
Comment (jorgehermo9) [reply website/cue/reference/remap/functions/unflatten.cue]: updated it in the last commit of this PR for simplicity

File: website/cue/reference/remap/functions/unflatten.cue [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+			description: "Whether to recursively unflatten the object values."
+			required:    false
+			default:     "true"
+			type: ["boolean"]
+		},
+	]
+	internal_failure_reasons: []
+	return: types: ["object"]
+
+	examples: [
```
Comment (pront) [inline website/cue/reference/remap/functions/unflatten.cue:34]: Worth adding an example [like this](https://playground.vrl.dev/?state=eyJwcm9ncmFtIjoiLiA9IHVuZmxhdHRlbiguKSIsImV2ZW50Ijp7ImEiOjMsImEuYiI6MiwiYS5jIjo0fSwiaXNfanNvbmwiOmZhbHNlLCJlcnJvciI6bnVsbH0%3D).

Comment (jorgehermo9) [reply website/cue/reference/remap/functions/unflatten.cue:34]: Thanks! You meant this case, right? https://github.com/vectordotdev/vrl/blob/5f635f93e7bfb060b747339e9a2e965741e0176d/src/stdlib/unflatten.rs#L60
Implemented in https://github.com/vectordotdev/vector/pull/21142/commits/2173ec398db6d095c298585327060421e2aa193e. I was not sure about the example's title

Or did you wanted to depict something like its done here: https://github.com/vectordotdev/vector/blob/beceef4a964a77acc3064c064eb27d265261c17d/website/cue/reference/remap/functions/unnest.cue#L57

Where it's shown that we can replace the original object with `. = unflatten(.)` 
(example of what I mean)
image


I've just noticed that in that `unnest` example, the `input` field that is set on the examples is not being rendered in the website, is this a bug?

**Pull request merged**